include/aws/common/string.h

```diff
@@ -68,27 +68,91 @@ AWS_STATIC_IMPL const uint8_t *aws_string_bytes(const struct aws_string *str) {
  * Returns true if bytes of string are the same, false otherwise.
  */
 AWS_STATIC_IMPL bool aws_string_eq(const struct aws_string *a, const struct aws_string *b) {
-    return a->len == b->len && !memcmp(a->bytes, b->bytes, a->len);
+    const uint8_t *a_ptr = a ? a->bytes : NULL;
```

Comment:
It seems like a handful of macros could reduce these to one-liners

#define STRING_TO_ARRAY_PARAMS(str) ((str) ? (str->bytes) : NULL), ((str) ? (str->len) : 0)
 
etc...

Reply:
I got rid of these lines in the latest versions